modules/dcache/src/main/java/diskCacheV111/services/PoolStatisticsV0.java

```diff
@@ -418,22 +360,25 @@ public void createDiffFile(File today, File yesterday, File resultFile) throws I
     }
 
     private class HourlyRunner implements Runnable {
-        private final Calendar _calendar;
-        private HourlyRunner(Calendar calendar) {
-            _calendar = calendar;
-            _nucleus.newThread(this,"FreeRunner").start();
+        private void updateCalendar() {
+            _calendar = new GregorianCalendar();
+            _calendar.set(Calendar.SECOND, 0);
         }
 
         @Override
         public void run() {
+            updateCalendar();
```

Comment:
1. Why we need to update the calendar on each run?
2. if we need it, the update happens in a different thread, thus access must be thread-safe

Reply:
I don't know if I got that right, but I think this run-method is called once an hour (by the executorService).
Then files are created with the current date and time in their name, so that the time has to be updated.
Or did i misunderstand your point?

Reply:
Ok. looks like it was required to print when the a time the task will run. You already use `scheduleAtFixedRate` which returns a `ScheduledFuture`. This is you store it, then you can query it:

```
scheduledTask =  executor.scheduleAtFixedRate(...);

...
long delayInMillis = scheduledTask.getDelay();
....
```